service/status-report/main.go

```diff
@@ -77,6 +85,43 @@ func main() {
 		klog.Exitf("Failed to get storageClient %q/%q: %v", storageClient.Namespace, storageClient.Name, err)
 	}
 
+	csvList := opv1a1.ClusterServiceVersionList{}
+	if err = cl.List(ctx, &csvList, client.InNamespace(storageClientNamespace)); err != nil {
+		klog.Exitf("Failed to list csv resources: %v", err)
+	}
+
+	var opVersion string
+	for idx := range csvList.Items {
+		candidate := &csvList.Items[idx]
+		if strings.HasPrefix(candidate.Name, csvPrefix) {
+			opVersion = candidate.Spec.Version.String()
+			break
+		}
+	}
```

Comment:
I believe we implemented somewhere GetCSVByPrefix(prefix) which we can use in this case as well.

But preferably I would like to see if we can implement a 
```
func Find[T any](list []T, func (item *T) bool) *T 
```
util in the utils package.

This will allow us to replace the code above with:
```
csv := find(csvList.Items, func(csv *opv1a1.ClusterServiceVersion) {
    return strings.HasPrefix(csv.Name, csvPrefix)
})
```

Reply:
done.